### PR TITLE
[ProgressBar] Add color prop

### PR DESCRIPTION
**Pull request by kaelig**
### WHY are these changes introduced?

This will fix an issue where a home card uses imperative language to describe the color prop's possible values in [this polaris-next fork of ProgressBar in web](https://github.com/Shopify/web/tree/f2ca3056929d56ae8f1c59b9fee9f8d68a378509/packages/%40shopify/polaris-next/components/ProgressBar).

This forked ProgressBar is used in two Home Cards, one of them being:

Screen Shot 2020-10-07 at 12 22 37 PM

I couldn't reproduce the other one, but the code shows it uses a "green" color:

https://github.com/Shopify/web/blob/master/app/sections/Home/HomeIndex/components/Onboarding/components/OnboardingFooter/components/OnboardingProgressBar/OnboardingProgressBar.tsx#L32

### WHAT is this pull request doing?

Adds a new prop using the new color system, that will help us remove this fork from polaris-next.

## 

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

Check out the new ProgressBar example in Storybook.

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

**Review by alex-page — APPROVED** [review]

File: src/components/ProgressBar/ProgressBar.scss [hunk 1]
```diff
@@ -40,11 +40,29 @@
   height: progress-bar-height(large);
 }
 
+.colorHighlight {
+  --p-progressbar-background: var(--p-surface-neutral, #{color('sky')});
+  --p-progressbar-indicator: var(--p-border-highlight, #{color('teal')});
+}
```
Comment (kaelig) [inline src/components/ProgressBar/ProgressBar.scss:46]: Not sure component-specific custom properties are the right way to fix this, guidance would be appreciated.

**Comment by kaelig** [comment]
Note: This isn't urgent, and will require a chat with designers to agree on whether this should make it into Polaris and if the colors chosen are appropriate.

File: src/components/ProgressBar/ProgressBar.tsx [hunk 2]
```diff
@@ -18,14 +19,24 @@ export interface ProgressBarProps {
    * @default 'medium'
    */
   size?: Size;
+  /**
+   * Size of progressbar
```
Comment (kyledurand) [inline src/components/ProgressBar/ProgressBar.tsx:23]: Looks like this was copy pasta from above

**Review by kyledurand — APPROVED** [review]
I like this approach 👍 Just make sure to change the prop description

File: src/components/ProgressBar/ProgressBar.tsx (hunk 2 above)
Comment (kaelig) [reply src/components/ProgressBar/ProgressBar.tsx]: Thanks! Fixed :)

**Comment by kaelig** [comment]
This last merge makes this change compatible with the default new design language! I'll ship it soon.

**Comment by kaelig** [comment]
FYI @whizkydee, it'd be great if you could make time to update the OnboardingTaskCard component over the next few weeks.

**Pull request merged**